File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void dfs(TreeNode* root, vector<int>& answer){
        if(!root) return; 

        dfs(root->left,answer);
        answer.push_back(root->val);
        dfs(root->right,answer); 
    }
    vector<int> inorderTraversal(TreeNode* root) {
        if(!root){
            return {}; 
        }

        vector<int> answer; 
        dfs(root,answer); 
        return answer; 
    }
};

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer> answer = new ArrayList<>(); 
    public List<Integer> inorderTraversal(TreeNode root) {
        if(root == null) return answer; 
        inorder(root);
        return answer; 
    }
    public void inorder(TreeNode root){
        if(root == null) return; 
        inorder(root.left);
        answer.add(root.val); 
        inorder(root.right); 
    }
}